demisto_sdk/commands/common/update_id_set.py

```diff
@@ -136,6 +137,60 @@ def get_changed_files(files_string):
     return added_files_list, modified_files_list, added_script_list, modified_script_list
 
 
+def build_dependency_graph(playbook_data):
+    """
+    Builds tasks flow graph.
+
+    Args:
+        playbook_data (dict): playbook yml data.
+
+    Returns:
+        DiGraph: all tasks of given playbook.
+    """
+    initial_task = playbook_data.get('starttaskid', '0')
+    tasks = playbook_data.get('tasks', {})
+
+    graph = networkx.DiGraph()
+    graph.add_node(initial_task, mandatory=True)  # add starting task as root of the direct graph
+
+    found_new_tasks = True
+    while found_new_tasks:
+        current_number_of_nodes = graph.number_of_nodes()
+        leaf_nodes = {node for node in graph.nodes() if graph.out_degree(node) == 0}
+
+        for leaf in leaf_nodes:
+            leaf_task = tasks.get(leaf)
+            leaf_mandatory = graph.nodes[leaf]['mandatory']
+
+            # In this case the playbook is invalid, one of the "nexttasks" fields contain invalid task id.
+            if not leaf_task:
+                print_error(f'{playbook_data.get("id")}: No such task {leaf} in playbook')
+                continue
+                # raise ValueError(f'No such task {leaf} in playbook {playbook_data.get("id")}')
+            leaf_next_tasks = sum(leaf_task.get('nexttasks', {}).values(), [])
```

Comment:
Not sure why you need to use sum on `nexttasks`

Reply:
in conditional tasks, there might be several lists:
![image](https://user-images.githubusercontent.com/30797606/87728136-1900e900-c7cb-11ea-82f9-a55025b6302e.png)